apis/v1alpha1/backendpolicy_types.go

```diff
@@ -0,0 +1,173 @@
+	//   name: myservice
+	//
+	// Otherwise, if the core API group is not desired, specify the desired
+	// group:
+	//
+	// fooRef:
+	//   group: acme.io
+	//   resource: foos
+	//   name: myfoo
+	//
+	// +optional
+	// +kubebuilder:default=core
+	Group string `json:"group"`
+
+	// Resource is the API resource name of the referent. Omitting the value
+	// or specifying the empty string indicates the services resource. For example,
+	// use the following to specify a services resource:
+	//
+	// fooRef:
+	//   name: myservice
+	//
+	// Otherwise, if the services resource is not desired, specify the desired
+	// group:
+	//
+	// fooRef:
+	//   group: acme.io
+	//   resource: foos
+	//   name: myfoo
+	//
+	// +optional
+	// +kubebuilder:default=services
+	Resource string `json:"resource"`
+
+	// Name is the name of the referent.
+	//
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// Port is the port of the referent. If unspecified, this policy applies to
+	// all ports on the backend.
+	// +optional
+	Port *int32 `json:"port,omitempty"`
+}
+
+// BackendTLSConfig describes TLS configuration for a backend.
+type BackendTLSConfig struct {
+	// ClientCertificateRef is a reference to a TLS client certificate that may
+	// be used to connect to these backends. If an entry in this list omits or
+	// specifies the empty string for both the group and the resource, the
+	// resource defaults to "secrets". An implementation may support other
+	// resources (for example, resource "mycertificates" in group
+	// "networking.acme.io").
+	//
+	// If a Secret is referenced, it must be of type "kubernetes.io/tls" and
+	// contain tls.crt and tls.key data fields that contain the certificate and
+	// private key to use for TLS.
+	//
+	// Support: Extended
+	//
+	// +optional
+	ClientCertificateRef *CertificateObjectReference `json:"clientCertificateRef,omitempty"`
+
+	// CertificateAuthorityRef is a reference to a resource that includes
+	// trusted CA certificates for the associated backends. If an entry in this
+	// list omits or specifies the empty string for both the group and the
+	// resource, the resource defaults to "secrets". An implementation may
+	// support other resources (for example, resource "mycertificates" in group
+	// "networking.acme.io").
+	//
+	// Support: Extended
+	//
+	// +optional
+	CertificateAuthorityRef *CertificateObjectReference `json:"certificateAuthorityRef,omitempty"`
+
+	// Options are a list of key/value pairs to give extended options to the
+	// provider.
+	//
+	// Support: Implementation-specific.
+	// +optional
+	Options map[string]string `json:"options,omitempty"`
+}
+
+// BackendPolicyStatus defines the observed state of BackendPolicy.
+type BackendPolicyStatus struct {
+	// Conditions describe the current conditions of the BackendPolicy.
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+// BackendPolicyConditionType is a type of condition associated with a
```

Comment:
Should `BackendPolicy` conditions follow the Gateway condition pattern being established in https://github.com/kubernetes-sigs/service-apis/pull/324?

Reply:
That's an interesting idea. When would this config be invalid for one backend and not another?

Reply:
Discussed a few more potential ways we might want to group conditions in another comment: https://github.com/kubernetes-sigs/service-apis/pull/268#discussion_r489191518

Reply:
Since there are so many potential ways to slice this up, I think the simplest for now is to have per Backend conditions at this level and then place implementation specific status on Routes (which already have per-Gateway status). I've updated the comments to suggest that approach here.